src/Sylius/Bundle/ReviewBundle/Validator/Constraints/UniqueUserEmailValidator.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\ReviewBundle\Validator\Constraints;
+
+use Sylius\Bundle\UserBundle\Doctrine\ORM\UserRepository;
+use Sylius\Component\User\Model\UserInterface;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+/**
+ * @author Mateusz Zalewski <mateusz.p.zalewski@gmail.com>
+ * @author Grzegorz Sadowski <grzegorz.sadowski@lakion.com>
+ */
+class UniqueUserEmailValidator extends ConstraintValidator
+{
+    /**
+     * @var UserRepository
+     */
+    private $userRepository;
+
+    /**
+     * @var TokenStorageInterface
+     */
+    private $tokenStorage;
+
+    /**
+     * @var AuthorizationCheckerInterface
+     */
+    private $authorizationChecker;
+
+    /**
+     * @param UserRepository $userRepository
+     * @param TokenStorageInterface $tokenStorage
+     * @param AuthorizationCheckerInterface $authorizationChecker
+     */
+    public function __construct(UserRepository $userRepository, TokenStorageInterface $tokenStorage, AuthorizationCheckerInterface $authorizationChecker)
```

Comment:
1. Hardcoded implementation, always use Interface.
2. Reviews should not know about User.